packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts

```diff
@@ -98,6 +98,12 @@ export function serverBareModulesPlugin(
           case "cloudflare-workers":
           case "deno":
             return undefined;
+          case "cloudflare-workers-esm": {
+            // If we're using Workers Modules (ESM), bundle everything except the special asset manifest import
+            if (path !== "__STATIC_CONTENT_MANIFEST") {
+              return undefined;
```

Comment:
I figured this was preferable to setting `external` in `compilerServer.ts`, but let me know.